### ISSUE TITLE
The jsx-closing-bracket-location rule for multi-line props

### ISSUE THREAD

**Issue by GreenGremlin** [yannickcr/eslint-plugin-react#2297]
The jsx-closing-bracket-location rule incorrectly warns for the following.

```jsx
<Page multiLineProp={{
  one: 1,
  two: 2,
}}> // The jsx-closing-bracket-location rule warns on this closing bracket.
  {content}
</Page>
```

The jsx-closing-bracket-location rule wants the closing bracket for the opening tag to be on the next line, like this.

```jsx
<Page multiLineProp={{
  one: 1,
  two: 2,
}}
>
  {content}
</Page>
```

This is the case for both line-aligned and tag-aligned. My opinion is that this should either be valid for both settings, or be exposed as a separate option.

Making the above example valid would result in including additional props before the closing tag also being valid. This may not be desirable for some, so it may be best for it to be a new option.

```jsx
<Page multiLineProp={{
  one: 1,
  two: 2,
}} otherProp="Hmmm...">
  {content}
</Page>
```

**Comment by ljharb** [comment]
I'm not sure why that _should_ be valid, to be honest.
